File: Python/Algoritmos_Orden/Quick.py
def selection(a):
    #recorre todo el arreglo
    for i in range(len(a)):
        #indice del elemento más pequeño
        small = i
        # encuentra el elemento más pequeño en el arreglo
        for j in range(i+1, len(a)):
            #compara el elemento más pequeño con el siguiente elemento
            if a[small] > a[j]: 
                #actualiza el indice del elemento más pequeño
                small = j
        #intercambia el elemento más pequeño con el primer elemento
        #intercambia los elementos
        a[i], a[small] = a[small], a[i]

def printArr(a):
    for i in range(len(a)):         #recorre todo el arreglo
        print (a[i], end = " ")     #imprime el elemento
a = [65, 26, 13, 23, 12]            #arreglo desordenado

print("Arrelo antes de ser ordenado: ")
printArr(a)
selection(a)
print("\nArreglo después de ser ordenado: ")
selection(a)
printArr(a)

File: JavaScript/Algoritmos_Orden/Quick.js
function swap(a, j, k) 
{

}

function partition(a, l, h) 
{

}

function qckSort(a, l, h) 
{
    if (l < h) {
        let pi = partition(a, l, h);
        qckSort(a, l, pi - 1);
        qckSort(a, pi + 1, h);
    }
}

let arreglo = [10, 7, 8, 9, 1, 5];
let largo = arreglo.length;

console.log("Antes de ordenar los elementos del array son: ");
for (const j of arreglo) {
    console.log(j + " ");
}
console.log();

qckSort(arreglo, 0, largo - 1)

console.log("Después de ordenar los elementos del array son: ");
for (let j = 0; j < largo; j++) {
    console.log(arreglo[j] + " ");
}
console.log();